test/models/concerns/user_multifactor_methods_test.rb

```diff
@@ -8,18 +8,31 @@ class UserMultifactorMethodsTest < ActiveSupport::TestCase
   end
 
   context "#mfa_enabled" do
-    should "return true if multifactor auth is not disabled" do
+    should "return true if multifactor auth is not disabled using OTP" do
       @user.enable_totp!(ROTP::Base32.random_base32, :ui_only)
 
       assert_predicate @user, :mfa_enabled?
     end
 
-    should "return true if multifactor auth is disabled" do
+    should "return false if multifactor auth is disabled" do
       @user.disable_totp!
 
       refute_predicate @user, :mfa_enabled?
     end
 
+    should "return true if multifactor auth is not disabled using WebAuthn" do
+      create(:webauthn_credential, user: @user)
+
+      assert_predicate @user, :mfa_enabled?
+    end
+
+    should "return true if multifactor auth is disabled using WebAuthn" do
+      create(:webauthn_credential, user: @user)
+      @user.webauthn_credentials.first.destroy
+
+      refute_predicate @user, :mfa_enabled?
+    end
+
```

Comment:
What's the purpose of having these tests? We have tests that confirm that the MFA level is set correctly when a webauthn credential is created/destroyed. https://github.com/rubygems/rubygems.org/blob/master/test/models/webauthn_credential_test.rb#L24

Reply:
These tests would have caught the problem with webauthn only users before the rake task. It also helps with the enforcement of MFA level being enabled and disabled only with devices (as you plan to add in a little bit). So, I figured it was worth adding these in here.